flow/templates/drexel-picotte.sh

```diff
@@ -12,7 +12,7 @@
     {% set nn = nn|default((nn_cpu, nn_gpu)|max, true) %}
     {% if partition == 'gpu' %}
 #SBATCH --nodes={{ nn|default(1, true) }}
-#SBATCH --ntasks-per-node={{ (gpu_tasks, cpu_tasks)|max }}
+#SBATCH --ntasks-per-node={{ ((gpu_tasks, cpu_tasks)|max / nn)|int }}
```

Comment:
This will only be correct if the number of nodes is evenly divisible by the number of tasks, right? Would this be an issue? If yes, should we protect against that?

Reply:
If the tasks cannot be evenly distributed across an integer number of nodes (e.g. `nrank=13`), then there is no way to request this with `--ntasks-per-node` - if you rounded up you would be providing the user with more ranks than they requested. Use `--ntasks={nranks}` instead in these cases.

Preserve the `--nodes=`, `--ntasks-per-node=`, request when tasks can be evenly distributed across nodes. It will provide a more efficient communication pattern.

Reply:
Yes, so I think we should raise an error in case that the requested configuration cannot be provisioned.

Reply:
We currently round in other templates for GPU partitions e.g. `expanse`. We could change that. Generally that won't matter since we take the ceiling and charges for GPU nodes are usually just for GPUs I if I understand correctly.

Reply:
Yes, they would likely be charged correctly. However, if the users launches their app with `srun` or `mpiexec` without arguments, then the autodetected number of tasks detected from the job configuration. If this is rounded up from what the user requested, then the user's script may fail (e.g. when it is coded to work with a specific number of ranks).

Reply:
Okay, in the case of GPU jobs does it make sense to have CPU tasks not be a multiple of GPUs? I feel that is something we should error at then.

Reply:
It is possible for systems to support a different number of CPU tasks and GPUs. For example, NCSA Delta does:
```
$ srun --account=bbgw-delta-gpu --partition=gpuA40x4 --tasks=7 --mem=48g --gpus=5 --pty zsh
$ echo $SLURM_NTASKS                                                                                                       
7
$ echo $SLURM_TASKS_PER_NODE
4,3
```
In this test, it assigned all 4 GPUs on the first node and 1 GPU on the 2nd.

Just because it is possible doesn't need that signac-flow needs to support it. I can't think of any reasonable workflows that would need this. Also, you would need to check each system separately whether it has configured SLURM to allow for this uneven task distribution.